crates/anemo/src/config.rs

```diff
@@ -229,6 +236,12 @@ impl Config {
     pub(crate) fn outbound_request_timeout(&self) -> Option<Duration> {
         self.outbound_request_timeout_ms.map(Duration::from_millis)
     }
+
+    pub(crate) fn shutdown_idle_timeout(&self) -> Duration {
+        self.shutdown_idle_timeout_ms
+            .map(Duration::from_millis)
+            .unwrap_or(Duration::MAX)
```

Comment:
This needs to reflect the documentation you included above and default to 1,000 milliseconds.

Reply:
I've actually amended the documentation and defaulted this to unlimited - I believe it's safer as default behaviour to have stronger guarantees rather the opposite.